src/main/java/org/gridsuite/study/server/networkmodificationtree/AbstractNodeRepositoryProxy.java

```diff
@@ -77,4 +81,41 @@ public void updateNode(AbstractNode node) {
     public void deleteAll(Set<UUID> collect) {
         nodeInfoRepository.deleteByIdNodeIn(collect);
     }
+
+    public Optional<String> getVariantId(UUID nodeUuid, boolean generateId) {
+        AbstractNode node = getNode(nodeUuid);
+        if (node.getType() == NodeType.ROOT) {
+            return Optional.of("");  // we will use the network initial variant
+        } else if (node.getType() == NodeType.NETWORK_MODIFICATION) {
+            NetworkModificationNode networkModificationNode = (NetworkModificationNode) node;
+            if (networkModificationNode.getVariantId() == null && generateId) {
+                networkModificationNode.setVariantId(UUID.randomUUID().toString());  // variant id generated with UUID format ????
+                updateNode(networkModificationNode);
+            }
+            return Optional.ofNullable(networkModificationNode.getVariantId());
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    public Optional<UUID> getModificationGroupUuid(UUID nodeUuid, boolean generateId) {
+        AbstractNode node = getNode(nodeUuid);
```

Comment:
Make different implementations in subclasses ?

Reply:
Done